### PR TITLE
[Term Entry] JavaScript Objects .groupBy() 

### PR DESCRIPTION
**Pull request by cecilezan**
### Description

- added the groupBy() method for Javascript
- Closes issue #4064 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+Title: '.groupBy()'
+Description: 'Groups objects by a key.'
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:3]:
```suggestion
Description: 'Used to group objects based on a key extracted from each object using a provided callback function.'
```
It is a best practice to be more specific.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+The **`.groupBy()`** method groups items according to the value returned by a [callback function](https://codecademy.com/ressources/docs/javascript/callbacks). The value of the `.groupBy()` method is a null-prototype object: it doesn't inherit any object methods.
+
+## Syntax
+
+```pseudo
+Object.groupBy(items, callfunc);
+```
+
+The `items` are usually an array. They will be grouped according to the result of the `callfunc`.
+The `callfunc` is a callback function. This function has to return a string or a symbol.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:23]:
```suggestion
- `items`: It is an array. They will be grouped according to the result of the `callfunc`.
- `callfunc`: It is a callback function. This function returns a string or a symbol.
```

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Example
+
+In this example, we want to group users by their subscription year.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:27]:
```suggestion
In the following example, the code groups users by their subscription year.
```
Don't use first and second-person pronouns.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  - 'Web Development'
+Tags:
+  - 'Methods'
+  - 'Objects'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+The **`.groupBy()`** method groups items according to the value returned by a [callback function](https://codecademy.com/ressources/docs/javascript/callbacks). The value of the `.groupBy()` method is a null-prototype object: it doesn't inherit any object methods.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:14]:
```suggestion
The **`.groupBy()`** method groups items according to the value returned by a [callback function](https://codecademy.com/ressources/docs/javascript/callbacks). The return value of the `.groupBy()` method is a null-prototype object that doesn't inherit any object methods.
```

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+The output would be:
+
+```shell
+{ 2020: [
+{ userName: 'Akim', yearSub: 2020 },
+{ userName: 'Lola', yearSub: 2020 },
+],
+  2022 : [
+{ userName: 'Lisa', yearSub: 2022 },
+]}
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:50]:
```suggestion
{
  '2020': [
    { userName: 'Akim', yearSub: 2020 },
    { userName: 'Lola', yearSub: 2020 }
  ],
  '2022': [ { userName: 'Lisa', yearSub: 2022 } ]
}
```
I removed some extra commas from the output.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+##
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:53]:
```suggestion
```

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    { userName: 'Lisa', yearSub: 2022 },
+    { userName: 'Akim', yearSub: 2020 },
+    { userName: 'Lola', yearSub: 2020 },
+    ];
+
+const subsByYear = Object.groupBy(subs, user => {
+    return user.yearSub;
+    });
+```
+
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:40]: Add a note saying that the reader needs to add a library like Lodash to run this code.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+The output would be:
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:41]:
```suggestion
The output of the above code will be:
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@cecilezan I mentioned some changes from my side, please make them.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md (hunk 7 above)
Comment (cecilezan) [reply content/javascript/concepts/objects/terms/groupBy/groupBy.md:40]: **Note:** To run this code, it is necessary to add a library like [Lodash](https://lodash.com/docs/4.17.15#groupBy).

**Review by cecilezan — COMMENTED** [review]
@avdhoottt I made the requested corrections

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md (hunk 7 above)
Comment (avdhoottt) [reply content/javascript/concepts/objects/terms/groupBy/groupBy.md:40]: @cecilezan You need to add that in the content and not as a comment. You can visit Codecademy's [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for that.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@cecilezan Visit our style guide and please make the changes.

File: content/javascript/concepts/objects/terms/groupBy/groupBy.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    { userName: 'Lisa', yearSub: 2022 },
+    { userName: 'Akim', yearSub: 2020 },
+    { userName: 'Lola', yearSub: 2020 },
+    ];
+
+const subsByYear = Object.groupBy(subs, user => {
+    return user.yearSub;
+    });
+```
+>**Note:** To run this code, it is necessary to add a library like [Lodash](https://lodash.com/docs/4.17.15#groupBy).
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/groupBy/groupBy.md:40]:
```suggestion
> **Note:** To run this code, it is necessary to add a library like [Lodash](https://lodash.com/docs/4.17.15#groupBy).
```

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review.

**Pull request merged**